### ISSUE TITLE
TipTap editor

### ISSUE THREAD

**Issue by danharrin** [laravel-filament/filament#13269]

**Comment by bzy107** [comment]
I think this is wonderful.
I was about to install the filament-tiptap-editor plugin, but I might not need to anymore.

So, I'd like to ask if this is intended to be used like the existing RichEditor?

For example:
```php
TipTapEditor::make('content')
```

**Comment by danharrin** [comment]
This is probably going to be implemented after v4 releases, please use the plugin in the meantime.

**Comment by danharrin** [comment]
Initial implementation in #15554

Todo for beta1:
- [x] Ability to inject JS and PHP extensions
- [x] Deferred file attachment uploads, only when form is submitted. With support for Spatie Media Library inc old file deletion and private storage.
- [x] Merge tags
- [x] Custom blocks

**Comment by binaryfire** [comment]
Not sure if this is already covered, but it'd be great if we can dynamically configure whatever toolbar buttons are included by default. Eg:

- What text tags are available (p, h1, h2 etc)
- Which fonts are shown in the dropdown (useful for CMSs where users can set global fonts)
- Color palette choices

etc

**Comment by balu-lt** [comment]
[TipTap v3](https://tiptap.dev/tiptap-editor-v3) BETA is out with awesome new features and is expected to be released later this year.

My most-wanted feature is Markdown support.

I'm keeping an eye on the upcoming releases, it'd be great to have this in Filament v4. Not sure what kind of breaking changes are in the major TipTap upgrade.

**Comment by binaryfire** [comment]
> TipTap v3 BETA is out with awesome new features and is expected to be released later this year.

I was going to post the same thing. More details on the changes: https://next.tiptap.dev/docs/resources/whats-new. They say the stable release is coming "later in 2025" so I'm assuming it's not far away. Providing it lands in time, I think it's worth using in Filament v4. 

All future TipTap development will be happening on v3. And I'm guessing all new third-party extensions will be made exclusively for v3.

**Comment by awcodes** [comment]
I'm sure filament will update at some point but given the lack of solid release dates I don't see filament including Tiptap v3 until a minor release after v4 is fully released.

**Comment by danharrin** [comment]
I am going to have a look at v3 and how stable it feels to me, I would like to include it in Filament 4 if it isn't going to cause too many problems, I don't want to be locked to v2 ideally.

**Comment by neverything** [comment]
@awcodes is it planned to have the Mention extension in the V4 release, or will we still be relying on your awesome plugin?

**Comment by awcodes** [comment]
@neverything i don't believe mentions are planned for core, i could be wrong, but also, my plugin will probably get depreciated with v4 and become plugins for the editor itself.
